packages/storybook/src/nimble/table-column/menu-button/table-column-menu-button.stories.ts

```diff
@@ -310,40 +353,29 @@ export const menuButtonColumn: StoryObj<MenuButtonColumnTableArgs> = {
         menuSlot: 'color-menu',
         menuRef: undefined,
         currentData: [...simpleData],
+        menuState: new MenuState(),
         updateMenuItems: (
             storyArgs: MenuButtonColumnTableArgs,
             e: CustomEvent<MenuButtonColumnToggleEventDetail>
         ): void => {
             if (e.detail.newState) {
                 const recordId = e.detail.recordId;
+                storyArgs.currentRecord = storyArgs.currentData.find(
+                    d => d.id === recordId
+                )!;
 
-                const changeFavoriteColor = (color: string): void => {
-                    const newData = storyArgs.currentData.map(d => {
-                        if (d.id !== recordId) {
-                            return d;
-                        }
-                        return {
-                            ...d,
-                            favoriteColor: color
-                        };
-                    });
-                    storyArgs.currentData = newData;
-                    void storyArgs.tableRef.setData(newData);
-                };
-
-                const item1 = document.createElement(menuItemTag);
-                item1.textContent = `[${recordId}] Blue`;
-                item1.addEventListener('change', () => changeFavoriteColor('Blue'));
-
-                const item2 = document.createElement(menuItemTag);
-                item2.textContent = `[${recordId}] Green`;
-                item2.addEventListener('change', () => changeFavoriteColor('Green'));
-
-                const item3 = document.createElement(menuItemTag);
-                item3.textContent = `[${recordId}] Purple`;
-                item3.addEventListener('change', () => changeFavoriteColor('Purple'));
-
-                storyArgs.menuRef.replaceChildren(item1, item2, item3);
+                const currentColor = storyArgs.currentRecord
+                    .favoriteColor as string;
```

Comment:
We can also create a tpye for the record data to avoid the as cast right?

```
        interface SimpleRecord extends TableRecord {
          id: string,
          firstName: string,
          lastName: string,
          favoriteColor: string;
        }
```